ledokku-bootstrap.sh

```diff
@@ -1,52 +1,187 @@
+
+# This script is used to configure the essentials of ledokku.
+## 1 => Check whether the program/application, "Whiptails" exists or not.
+## 2 => Making sure that the script is runing with root permissions.
+## 3 => Update and Upgrade the VPS.
+## 4 => Check whether the program/application, "Dokku" was Installed or not.
+## 5 => Upgrading dokku to the latest version.
+## 6 => Check Whether Plugins are installed or not (Redis, Postgres)
+## 7 => Install Redis and Postgress, if they were not installed
+## 7 => Install Ledokku
+## ? => Update Script (Will be added later)
+
 set -e
 
-# Check that dokku is installed on the server
-ensure-dokku() {
-  if ! command -v dokku &> /dev/null
-  then
-      echo "dokku is not installed, please follow our getting started guide first"
-      echo "https://www.ledokku.com/docs/getting-started"
+function define-colors {
+
+  # Using tput will eliminate the usage of "-e" in echo, and can be used anywhere
+  ## Color Palet
+  ## Should exist in every script
+  RED="$(tput setaf 1)" # ${RED}
+  GREEN="$(tput setaf 2)" # ${GREEN}
+  YELLOW="$(tput setaf 3)" # ${YELLOW}
+  BLUE="$(tput setaf 123)" # ${BLUE}
+  END="$(tput setaf 7)" # ${END
+}
+
+function system-info() {
+  
+  # Finding Information about the server
+  ## Basic VPS info
+
+  DOKKU_SSH_HOST=$(curl -4 ifconfig.co)
+  OS=$( $(compgen -G "/etc/*release" > /dev/null) && cat /etc/*release | grep ^NAME | tr -d 'NAME="' || echo "${OSTYPE//[0-9.]/}")
+}
+
+function check-whiptail() {
+  
+  # Checking if whiptail is available or not
+  if which whiptail >/dev/null; then
+      echo "${GREEN}whiptail exists${END}"
+      # Continue the script
+  else
+      echo "${RED}whiptail does not exist${END}"
+      echo "Install whiptail and re-run the script, your OS is ${OS}"
+      exit
+      # As I already know the OS, I can also automate this process, but will save it for later.
+      # This is just a matter of finding all the possible OS people might use and writing an if statement.
+  fi
+}
+
+function check-root() {
+  
+  # Check root and if not root take permissions (Some providers doesnot support passwordless sudo)
+  ## It is always better to do this.
+  ## We will not face any further issues, during any sort of compulsory sudo commands; like the case for installing plugins in Dokku or Giving permissions to our scripts
+  if [ "$(whoami)" == "root" ] ; then
+      echo "${YELLOW}Nice you are running the script as root!${END}"
+  else
+      echo "${RED}Please! Run the script with root access${END}, without root access I cannot create plugins in Dokku"
       exit
   fi
 }
 
+function system-update {
+
+  # Updating and Upgrading system
+  ## If dokku was installed, it will be automatically updated to the latest version
+  ## Staying up to date is always good
+  if (whiptail --title "Update and Upgrade System " --yes-button "Yes" --no-button "Skip"  --yesno "Do you wish to Update packges and Upgrade your system?" 10 60) then
```

Comment:
@akhil-naidu 

Firstly, *packages typo. 

And secondly, could you please make the message more verbose, so it is clear that we are talking about the packages related to dokku/ledokku. Otherwise first trying this out I was not sure what kind of upgrades are we talking about here.

Reply:
It is updating APT(Advanced Package Tool) => nothing but `sudo apt update`

Edit 02: I will update the typo and will be more specific.

Reply:
Let me know if there are any issues in the flow of the logic. Meanwhile, I will check for typos and other instructions.

Reply:
@akhil-naidu Apart from typos everything else looks good to me and just worked on a fresh Digital Ocean server.